### ISSUE TITLE
deploy resource creates recursive symbolic links

### ISSUE THREAD

**Issue by BackSlasher** [chef/chef#2702]
TL;DR: Chef runs `run_symlinks_before_migrate`, causing symlinked directories to contain symlinks to themselves

Consider this:

``` bash
mkdir /tmp/bla{,/a}
ln -fs /tmp/bla/{a,b}
```

now `ls -l /tmp/bla{,/a}` prints:

``` text
/tmp/bla:
total 4
drwxrwxr-x 2 back slash 4096 dec 24 16:14 a
lrwxrwxrwx 1 back slash   10 dec 24 16:15 b -> /tmp/bla/a

/tmp/bla/a:
total 0
```

now execute `ln -fs /tmp/bla/{a,b}` again, and check  `ls -l /tmp/bla{,/a}`:

```
/tmp/bla:
total 4
drwxrwxr-x 2 back slash 4096 dec 24 16:16 a
lrwxrwxrwx 1 back slash   10 dec 24 16:15 b -> /tmp/bla/a

/tmp/bla/a:
total 0
lrwxrwxrwx 1 back slash 10 dec 24 16:16 a -> /tmp/bla/a
```

Same happens with Ruby's `FileUtils.ln_sf`.
The method `run_symlinks_before_migrate` is invoked twice, once from `migrate` and once from `link_tempfiles_to_current_release`.
I tried removing the one from `link_tempfiles_to_current_release` (because it's called after the migration, and the method _is_ called "before_migrate").
Seemed to work OK for me, but it failed rspec tests.
What do?
